### PR TITLE
Add more channel contracts. Fix a bug

### PR DESCRIPTION
**Pull request by swagatmishra2007**
- Added some more api's for new channels like line and alexa
- Added api's to regenerate keys for directline
- Whitespaces are allowed in auth connection names, so updated the regex there

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by swagatmishra2007** [comment]
A lot of the errors seem to be unrelated to my changes? (probably existing from before?)

**Comment by swagatmishra2007** [comment]
Does making a regex more relaxed (allowing whitespace in connection parameter change) count as a breaking change and need a version rev?

**Comment by markcowl** [comment]
@swagatmishra2007  The required checks must pass, or have a good reason for not passing.  The public.rest-api-specs checks are generally required, and can be easily fixed by following the advice in the logs and instructions.

Please fix these issues.  Rerunning the checks to refresh logs

File: specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json [hunk 1]
```diff
@@ -2666,10 +2881,37 @@
...
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "WebChatChannel",
+        "DirectLineChannel"
+      ],
+      "x-ms-enum": {
+        "name": "channelName",
+        "modelAsString": false,
```
Comment (markcowl) [inline specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json:2904]: These values will *never* change?  If there is a chance these values will change ina  future api-version, this should use modelAsString: true

**Review by markcowl — CHANGES_REQUESTED** [review]
Need to fix azure-rest-api-specs validations as well.

**Comment by markcowl** [comment]
@swagatmishra2007 Please respond to review comments.  Will close this at EOW if there is no further activity.

**Pull request closed without merging**